apollo-sample/src/main/java/com/example/apollographql/sample/MainActivity.java

```diff
@@ -23,39 +27,60 @@
-    SampleApplication application = (SampleApplication) getApplication();
-    application.githuntApiService()
-        .githuntFeed(new OperationRequest<>(
-            new FeedQuery(FeedQuery.Variables.builder()
-                .limit(10)
-                .type(FeedType.HOT)
-                .build())))
+    final RecyclerView responses = (RecyclerView) findViewById(R.id.responses);
+
+    final SampleApplication application = (SampleApplication) getApplication();
+
+    final Subject<Integer> upvoteObserver = PublishSubject.create();
+
+    final PostsAdapter adapter = new PostsAdapter(upvoteObserver);
+    responses.setAdapter(adapter);
+    responses.setLayoutManager(new LinearLayoutManager(this));
+
+    upvoteObserver.subscribe(new SimpleObserver<Integer>() {
+      @Override public void onNext(Integer postId) {
+        final Upvote.Variables variables = Upvote.Variables.builder()
+            .postId(postId)
+            .build();
+
+        application.postsService()
+            .upvote(new Upvote(variables))
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe(new SimpleObserver<Response<Upvote.Data>>() {
+              @Override public void onNext(Response<Upvote.Data> response) {
+                adapter.upvotePost(response.data().upvotePost());
+              }
+            });
+      }
+    });
+
+    application.postsService()
+        .allPosts(new AllPosts())
         .subscribeOn(Schedulers.io())
         .observeOn(AndroidSchedulers.mainThread())
-        .subscribe(new Observer<Response<FeedQuery.Data>>() {
-          @Override public void onSubscribe(Disposable d) {
-          }
+        .subscribe(new SimpleObserver<Response<AllPosts.Data>>() {
+          @Override public void onNext(Response<AllPosts.Data> response) {
+            txtResponse.setVisibility(View.GONE);
+            responses.setVisibility(View.VISIBLE);
 
-          @Override public void onNext(Response<FeedQuery.Data> response) {
-            StringBuffer buffer = new StringBuffer();
-            for (FeedQuery.Data.Feed feed : response.data().feed()) {
-              buffer.append("name:" + feed.repository().fragments().repositoryFragment().name());
-              buffer.append(" owner: " + feed.repository().fragments().repositoryFragment().owner().login());
-              buffer.append(" postedBy: " + feed.postedBy().login());
-              buffer.append("\n~~~~~~~~~~~");
-              buffer.append("\n\n");
-            }
-            txtResponse.setText(buffer.toString());
+            adapter.allPosts(response.data().posts());
           }
+        });
+  }
 
-          @Override public void onError(Throwable e) {
-            Log.e(TAG, "", e);
-            Toast.makeText(MainActivity.this, "onError(): " + e.getMessage(), Toast.LENGTH_LONG)
-                .show();
-          }
+  private abstract class SimpleObserver<T> implements Observer<T> {
```

Comment:
I'd prefer to exclude this as it makes sample more complicated. I feel like some users might think that this part is a requirement.